tests/components/openweathermap/test_config_flow.py

```diff
@@ -0,0 +1,216 @@
+            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        )
+
+        await hass.async_block_till_done()
+
+        conf_entries = hass.config_entries.async_entries(DOMAIN)
+        entry = conf_entries[0]
+        assert entry.state == "loaded"
+
+        await hass.config_entries.async_unload(conf_entries[0].entry_id)
+        await hass.async_block_till_done()
+        assert entry.state == "not_loaded"
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == CONFIG[CONF_NAME]
+        assert result["data"][CONF_LATITUDE] == CONFIG[CONF_LATITUDE]
+        assert result["data"][CONF_LONGITUDE] == CONFIG[CONF_LONGITUDE]
+        assert result["data"][CONF_API_KEY] == CONFIG[CONF_API_KEY]
+
+
+async def test_form_import(hass):
+    """Test we can import yaml config."""
+    mocked_owm = _create_mocked_owm(True)
+
+    with patch(
+        "homeassistant.components.openweathermap.config_flow.OWM",
+        return_value=mocked_owm,
+    ), patch(
+        "homeassistant.helpers.update_coordinator.DataUpdateCoordinator.async_refresh",
+        _async_refresh,
+    ), patch(
+        "homeassistant.components.openweathermap.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.openweathermap.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=VALID_YAML_CONFIG.copy(),
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["data"][CONF_LATITUDE] == hass.config.latitude
+        assert result["data"][CONF_LONGITUDE] == hass.config.longitude
+        assert result["data"][CONF_API_KEY] == VALID_YAML_CONFIG[CONF_API_KEY]
+
+        await hass.async_block_till_done()
+        assert len(mock_setup.mock_calls) == 1
+        assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_options(hass):
+    """Test that the options form."""
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, unique_id="openweathermap_unique_id", data=CONFIG
+    )
+    config_entry.add_to_hass(hass)
+
+    mocked_owm = _create_mocked_owm(True)
+
+    with patch(
+        "homeassistant.components.openweathermap.config_flow.OWM",
+        return_value=mocked_owm,
+    ), patch(
+        "homeassistant.helpers.update_coordinator.DataUpdateCoordinator.async_refresh",
+        _async_refresh,
+    ):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert config_entry.state == "loaded"
+
+    with patch(
+        "homeassistant.components.openweathermap.config_flow.OWM",
+        return_value=mocked_owm,
+    ), patch(
+        "homeassistant.helpers.update_coordinator.DataUpdateCoordinator.async_refresh",
+        _async_refresh,
```

Comment:
This shouldn't be needed. If it is, please patch the library instead of DataUpdateCoordinator